### PR TITLE
Ruby Tic-Tac-Toe: Game_logic

### PR DESCRIPTION
**Pull request by Temmarie**
In this third milestone, we were required to add real logic for our game, removing all inline comments used in the previous milestone. We have covered all edge cases, like :
- Asking the players if they'll like to restart the game,
- Alerting the players when they pick an already taken cell,
- Alerting the players when they input an invalid character.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,91 +1,105 @@
...
+  end
+  puts
+  print "Player Two press \'P'\ to enter name or \'Enter'\ to play as guest: "
+  id2 = gets.chomp.upcase
+  if id2 == 'P'
+    print 'Player Two name please: '
+    players2 = gets.chomp
+  else
+    players2 = 'Guest__02'
+  end
```
Comment (maelfosso) [inline bin/main.rb]: **`[OPTIONAL]`** Can be asked once. So, please, could you move it out of that loop and just ask that at the beginning?

File: bin/main.rb [hunk 2]
```diff
@@ -1,91 +1,105 @@
...
+
+          if board.positions.include?(input)
+            playa.player_input << inputs
+            board.positions[inputs] = playa.char
+            break
+          else
+            puts 'Invalid move/character try again!'
+            puts
+            next
+          end
```
Comment (maelfosso) [inline bin/main.rb]: 1. **Check** the input validity into a method from `Board` class
```ruby
def input_valid?(input)
...
return true/false
end
```

2. **Update the board** into the method from a `Board` class
```ruby
def update_board
@positions[inputs] = current_player.char
end
```

File: bin/main.rb [hunk 3]
```diff
@@ -1,91 +1,105 @@
...
+        puts board.display_board
+        puts
+      end
+      if board.won?(playa.player_input)
+        puts "#{playa.name} Wins!!!"
+
+        game_on = false
+        break
+
+      elsif board.positions.none?(Integer)
```
Comment (maelfosso) [inline bin/main.rb]: As a method, please.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @Temmarie @OA7 
Everything is working :+1: Well job done !!
However, there are a few things you should change related to the SOLID principle.

Please, read the comments left below and make the suggested changes then submit a new code review request

Thanks
Mael FOSSO

**Review by suretrust — APPROVED** [review]
Hi team, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged! 👍 

_Good luck with your future milestones and happy coding!_ 🎉

**Pull request merged**